src/Customer/Service.php

```diff
@@ -31,29 +31,29 @@ public function find(string $uid): Customer;
     /**
      * Associates a card to a customer.
      *
-     * @param Customer $customer
-     * @param Card $card
+     * @param string $customerUid
+     * @param string $cardUid
      *
      * @return Card
      */
-    public function addCard(Customer $customer, Card $card): Card;
+    public function addCard(string $customerUid, string $cardUid): Card;
 
     /**
      * Deletes a customers card.
      *
-     * @param Customer $customer
-     * @param Card $card
+     * @param string $customerUid
+     * @param string $cardUid
      *
      * @return void
      */
-    public function deleteCard(Customer $customer, Card $card): void;
+    public function deleteCard(string $customerUid, string $cardUid): void;
```

Comment:
Why not using the objects anymore? Objects (even data objects) should always be preferred from primitive types like string.

Reply:
Because we'll have to find those object in order to call those method which mean another call to the Stripe API. Whereas Stripe API just needs uids.